### PR TITLE
du: directories have apparent size of 0

### PR DESCRIPTION
**Pull request by shinhs0506**

### CODE REVIEW

File: tests/by-util/test_du.rs [hunk 1]
```diff
@@ -580,97 +580,62 @@ fn test_du_invalid_threshold() {
...
-    let result = ts.ucmd().arg("--apparent-size").succeeds();
+    let (at, mut ucmd) = at_and_ucmd!();
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
+    at.mkdir_all("a/b");
+    at.touch("a/b/file1");
+    at.touch("a/b/file2");
+
+    at.write("a/b/file1", "foo");
+    at.write("a/b/file2", "foobar");
```
Comment (cakebaker) [inline tests/by-util/test_du.rs:590]: Using `touch` is not necessary, `write` creates the file if it doesn't exist.

File: tests/by-util/test_du.rs [hunk 2]
```diff
@@ -580,97 +580,62 @@ fn test_du_invalid_threshold() {
...
 #[test]
 fn test_du_bytes() {
-    let ts = TestScenario::new(util_name!());
-    let result = ts.ucmd().arg("--bytes").succeeds();
+    let (at, mut ucmd) = at_and_ucmd!();
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
+    at.mkdir_all("a/b");
+    at.touch("a/b/file1");
+    at.touch("a/b/file2");
```
Comment (cakebaker) [inline tests/by-util/test_du.rs:617]: See my other comment

File: tests/by-util/test_du.rs [hunk 3]
```diff
@@ -580,97 +580,62 @@ fn test_du_invalid_threshold() {
...
+    let result = ucmd.args(&["--apparent-size", "--all", "a"]).succeeds();
+
+    #[cfg(not(target_os = "windows"))]
     {
-        let result_reference = unwrap_or_return!(expected_result(&ts, &["--apparent-size"]));
-        assert_eq!(result.stdout_str(), result_reference.stdout_str());
+        result.stdout_contains("1\ta/b/file1");
+        result.stdout_contains("1\ta/b/file2");
+        result.stdout_contains("1\ta/b");
+        result.stdout_contains("1\ta");
```
Comment (cakebaker) [inline tests/by-util/test_du.rs:599]: I think the patterns should end with `\n` as I guess you want to verify whether there are lines containing "1\ta/b" and "1\ta".

File: tests/by-util/test_du.rs [hunk 4]
```diff
@@ -580,97 +580,62 @@ fn test_du_invalid_threshold() {
...
-22\t./subdir
-26\t.
-"
-    );
+    #[cfg(target_os = "windows")]
+    {
+        result.stdout_contains("1\ta\\b\\file1");
+        result.stdout_contains("1\ta\\b\\file2");
+        result.stdout_contains("1\ta\\b");
+        result.stdout_contains("1\ta");
```
Comment (cakebaker) [inline tests/by-util/test_du.rs:607]: See my other comment.

**Comment by cakebaker** [comment]
When I run the updated tests, without doing any changes to `du.rs`, all tests pass. Is that intentional?

**Comment by shinhs0506** [comment]
@cakebaker everything resolved, thanks :)

**Comment by cakebaker** [comment]
Thanks! Good idea of introducing `stdout_contains_line`, it's a much cleaner solution than what I suggested.

**Pull request merged**